cli/azd/pkg/ext/models.go

```diff
@@ -0,0 +1,32 @@
+package ext
+
+type ScriptType string
+type ScriptLocation string
+
+const (
+	ScriptTypeBash       ScriptType     = "bash"
+	ScriptTypePowershell ScriptType     = "powershell"
+	ScriptLocationInline ScriptLocation = "inline"
+	ScriptLocationPath   ScriptLocation = "path"
+)
+
+type ScriptConfig struct {
+	// Internal name of the hook running for a given command
+	Name string `yaml:",omitempty"`
+	// The type of script hook (bash or powershell)
+	Type ScriptType `yaml:"type,omitempty"`
+	// The location of the script hook (file path or inline)
+	Location ScriptLocation `yaml:"location,omitempty"`
+	// When location is `path` a file path must be specified relative to the project or service
+	Path string `yaml:"path,omitempty"`
+	// When location is `inline` a script must be defined inline
```

Comment:
If we're always going to evaluate on `relative` path, what do you think about following the github actions model, and have a single `Run` parameter that the user would place either an inline script, or a relative path.

The option will be inferred from the input, similar to github [actions](https://docs.github.com/en/actions/learn-github-actions/essential-features-of-github-actions#adding-scripts-to-your-workflow).

The familiarity here will help ease into an authoring experience.

Reply:
Lets keep this simple for now and move forward with the separation of path vs script and then we can get some additional feedback from the community.